chat.js

```diff
@@ -350,6 +353,38 @@ class CommandContext {
 		}
 		return true;
 	}
+	checkR9k(user, room, message) {
+		let checksum = Chat.md5(toId(message)).substr(0, 16);
```

Comment:
We tried testing with a room and it had ~3 million messages since 2013 in total, so I don't see this being too bad of a problem.